checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -1385,6 +1385,41 @@ && varTrackedInObligations(obligations, (LocalVariableNode) receiver))
       checkEnclosingMethodIsCreatesMustCallFor(node, enclosingMethodTree);
     }
 
+    // Special case: re-assigning a field to a value in an owning position on the RHS is
+    // permitted. For example, if the field is a class whose constructor takes another instance
+    // of itself (such as a node in a linked list) in an owning position, re-assigning the field
+    // to a new instance that takes the field's value as an owning parameter is safe (the new value
+    // has taken responsibility for closing the old value). In that case, it is safe to skip the
+    // must call vs called methods check below (but an @CreatesMustCallFor annotation is still
+    // required, so this special case must be here).
```

Comment:
I find this comment to be a bit hard to parse.  Let me write what I think is going on and then you can decide how to edit / integrate.  The special case we are handling here is where the field being assigned is itself getting passed in an owning position to another method on the RHS of the assignment, e.g.:
```java
Foo takesOwnership(@Owning Foo foo) { ... }
...
@CreatesMustCallFor("this")
void overwrite() {
  this.f = takesOwnership(f);
} 
```
In such a case, it is not required that the must-call obligation of the field be satisfied via method calls before the assignment, since the invoked method will take ownership of the object previously referenced by the field and handle the obligation.

Taking a step back, I wish we didn't need yet another special case here.  But, I can't immediately think of a significantly cleaner solution, so we can go with this for now.